src/test/java/automationpractice/com/features/RegisterFeatures.java

```diff
@@ -0,0 +1,79 @@
+package automationpractice.com.features;
+
+import automationpractice.com.pages.HeaderBox;
+import automationpractice.com.pages.HomePage;
+import automationpractice.com.pages.LoginPage;
+import automationpractice.com.pages.RegistrationPage;
+import cucumber.api.java.en.Given;
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+import org.apache.commons.lang3.RandomStringUtils;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class RegisterFeatures {
+
+    final private String randomString = RandomStringUtils.randomAlphanumeric(13);
+    final private String randomEmail = String.format("%s@gmai.com", randomString);
+
+    private HomePage homePage;
+    private HeaderBox headerBox;
+    private LoginPage loginPage;
+    private LoginPage email;
+    private RegistrationPage registrationPage;
+
+
+    @Given("^User is on home page$")
+    public void userIsOnHomePage() {
+        homePage.open();
+    }
+
+
+    @When("^he types unique e-mail$")
+    public void heTypesUniqueEMail() {
+        loginPage.typeEmailRegistration(randomEmail);
+    }
```

Comment:
Między tymi krokami brakuje przejścia na stronę login page. Żeby przejśc na stronę login page trzeba kliknąć najpierw na sign in button w headerze. Tutaj powinieneś mieć Given z "he is on login page" i When z "he submits form with unique email"